src/rules/selector-max-specificity/index.js

```diff
@@ -35,15 +35,21 @@ export default function (max) {
         resolvedNestedSelector(selector, rule).forEach(resolvedSelector => {
           // Return early if selector contains a not pseudo-class
           if (selector.indexOf(":not(") !== -1) { return }
+          // Return early if selector contains a matches
+          if (selector.indexOf(":matches(") !== -1) { return }
```

Comment:
I don't think the above checks are safe, because they are not going to know if the string occurs within a comment or string. My suggestion is that we drop them and rely on the try-catch.